numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -328,29 +328,45 @@ gentype_nonzero_number(PyObject *m1)
 static PyObject *
 gentype_str(PyObject *self)
 {
-    PyObject *arr, *ret = NULL;
+    PyObject  *item, *item_str;
 
-    arr = PyArray_FromScalar(self, NULL);
-    if (arr != NULL) {
-        ret = PyObject_Str((PyObject *)arr);
-        Py_DECREF(arr);
+    item = gentype_generic_method(self, NULL, NULL, "item");
+    if (item == NULL) {
+        return NULL;
     }
-    return ret;
-}
 
+    item_str = PyObject_Str(item);
+    Py_DECREF(item);
+    return item_str;
+}
 
 static PyObject *
 gentype_repr(PyObject *self)
 {
-    PyObject *arr, *ret = NULL;
+    PyObject *item, *item_str;
 
-    arr = PyArray_FromScalar(self, NULL);
-    if (arr != NULL) {
-        /* XXX: Why are we using str here? */
-        ret = PyObject_Str((PyObject *)arr);
-        Py_DECREF(arr);
+    item = gentype_generic_method(self, NULL, NULL, "item");
+    if (item == NULL) {
+        return NULL;
     }
-    return ret;
+
+    item_str = PyObject_Repr(item);
+    Py_DECREF(item);
+    return item_str;
+}
+
+static PyObject *
```

Comment:
I think this is identical to `gentype_str`, no?  So, one could just use the latter below.